src/client.js

```diff
@@ -1764,6 +1768,44 @@ class Client {
         return result;
     }
 
+    /**
+     * This is the exposed/public method to request context data for a specific report.
+     * @param {*} callContext: {reportName: string, context: string, selection: string, selectionCount: number, formData: any}
+     * @param {string} representation the expected representation ('xml', 'BadgerFish', or 'SimpleJson'). If not set, will use the current representation
+     *
+     * @returns {Campaign.ReportContext} an object containing the context data for a specific report
+     */
+    async getReport(callContext, representation) {
+        try {
+            if(callContext.formData?.ctx) {
+                var xmlCtx = this._fromRepresentation('ctx', callContext.formData.ctx);
+                var formData = {...callContext.formData, ctx: DomUtil.toXMLString(xmlCtx)};
+            }
+            const request = {
+                url: `${this._connectionParameters._endpoint}/report/${callContext.reportName}?_noRender=true&_context=${callContext.context}&_selection=${callContext.selection}&_selectionCount=${callContext.selectionCount}`,
+                headers: {
+                    'X-Security-Token': this._securityToken,
+                    'Cookie': '__sessiontoken=' + this._sessionToken, 
+                    'Content-Type': 'application/x-www-form-urlencoded'
+                },
+                method: 'POST',
+                credentials: 'include',
+                data : qs.stringify(formData)
+            };
+            
+            for (let h in this._connectionParameters._options.extraHttpHeaders)
+                request.headers[h] = this._connectionParameters._options.extraHttpHeaders[h];
+            const body = await this._makeHttpCall(request);
+            const xml = DomUtil.parse(body);
+            const result = this._toRepresentation(xml, representation);
+            return result;
+        } catch(ex) {
+            if (this._traceAPICalls)
```

Comment:
This should have already been logged by _makeHttpCall, why do you need to log again?